File: Arrays/Basics/ARRAY_WeiredArray.cpp
#include<iostream>
#include<vector>
using namespace std;

int weired_array(int n, int arr[])
{
    if(n<2)
    {
        return 0;
    }

    vector<int> aar;
    int cnt=0;
    for(int i=0; i<n; i++)
    {
        for(int j=0; j<n; j++)
        {
            if(arr[i]==arr[j])
            cnt++;
            else{
                aar.push_back(cnt);
                cnt=0;
                break;
            }
        }

    }

    int maxm = 0;
    for(int x:aar)
    {
        if(x>maxm)
        {
            maxm = x;
        }
    }
    
    return maxm;
}


int main()
{

    int n;
    cin>>n;

    int arr[n];
    for(int i=0; i<n; i++)
    cin>>arr[i];

    cout<<weired_array(n,arr)<<endl;

    // vector<int>arr1 = {3,3,1,3,3};
    // cout<<weired_array(arr1)<<endl;

    // vector<int>arr2 = {5};
    // cout<<weired_array(arr2)<<endl;

    

    return 0;
}

File: Arrays/Basics/ARRAY_WeiredArray.java
import java.util.Arrays;

public class ARRAY_WeiredArray {
    public static void main(String[] args) {
        int a[] = { 3, 3, 1, 3, 3 };
        System.out.println(weiredArray(a));
    }

    public static int weiredArray(int[] a) {
        if (a.length < 2) {
            return 0;
        }
        int[] li = new int[a.length];
        int cnt = 0;
        int maxm = 0;

        for (int i = 0; i < a.length; i++) {
            for (int j = i; j < a.length; j++) {
                if (a[i] == a[j]) {
                    cnt++;
                } else {
                    li[i] = cnt;
                    cnt = 0;
                    break;
                }
            }

        }

        maxm = Arrays.stream(li).max().getAsInt();
        return maxm;
    }
}